### PR TITLE
Test added for accessing ArrayItem from struct instance member

### PR DESCRIPTION
**Pull request by HarshitaKalani**
Fix: https://github.com/lfortran/lfortran/issues/1734#issuecomment-1626886324

### CODE REVIEW

File: integration_tests/arrayitem_01.f90 [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module arrayitem_module_01
+    type :: block_1
+        integer :: a, c(5)
+    end type
+    type(block_1) :: struct_instance_block_1
+end module
+
+program arrayitem_01
+    use arrayitem_module_01
+    struct_instance_block_1%c(1) = 19
```
Comment (Pranavchiku) [inline integration_tests/arrayitem_01.f90:10]: Please add `if (struct_instance_block_1%c(1) /= 19) error stop` to ensure correctness of output

**Review by Pranavchiku — APPROVED** [review]
I left a comment, rest looks good to me.

**Review by Pranavchiku — APPROVED** [review]
LGTM! Thanks!

**Comment by certik** [comment]
Thanks!

**Pull request merged**